crates/corro-agent/src/agent/util.rs

```
@@ -1271,57 +1279,109 @@ pub fn process_complete_version<T: Deref<Target = rusqlite::Connection> + Commit
+        .query_map(params, |row| Ok((row.get(0)?, row.get(1)?, row.get(2)?)))?
+        .collect::<rusqlite::Result<Vec<(CrsqlDbVersion, CrsqlSeq, i64)>>>()?;
+
+    // Ignore this, it's just for debugging if we hit this very rare VDBE bug in production
+    let last_rowids_len = last_rowids.len();
+    if last_rowids_len != len {
+        // This should never happen, but if it does, i need data for debugging
+        let query_plan = sp
+            .prepare(&format!("EXPLAIN QUERY PLAN {INSERT_CRSQL_CHANGES_QUERY}",))?
+            .query_map(params, |row| {
+                Ok((row.get(0)?, row.get(1)?, row.get(2)?, row.get(3)?))
+            })?
+            // id, parent, notused, detail
+            .collect::<rusqlite::Result<Vec<(i32, i32, i32, String)>>>()?;
+        let vdbe_program = sp
+            .prepare(&format!("EXPLAIN {INSERT_CRSQL_CHANGES_QUERY}"))?
+            .query_map(params, |row| {
+                Ok((
+                    row.get(0)?,
+                    row.get(1)?,
+                    row.get(2)?,
+                    row.get(3)?,
+                    row.get(4)?,
+                    row.get(5)?,
+                    row.get(6)?,
+                    row.get(7)?,
+                ))
+            })?
+            .collect::<rusqlite::Result<
+                Vec<(
+                    i32,            // addr
+                    String,         // opcode
+                    i32,            // p1
+                    i32,            // p2
+                    i32,            // p3
+                    Option<String>, // p4
+                    Option<i32>,    // p5
+                    Option<String>, // comment
+                )>,
+            >>()?;
+        let details = json!({
+            "changes_len": len,
+            "returned_rowids_len": last_rowids.len(),
+            "query_plan": query_plan,
+            "vdbe_program": vdbe_program,
+        });
+        error!("Possible returning statement BUG: {details}");
```

Would we want to return an error here so we rollback the savepoint?